src/common/PropertyExt.js

```diff
@@ -154,16 +154,36 @@
     };
 
     // Publish Properties  ---
-    PropertyExt.prototype.publishedProperties = function (includePrivate) {
+    PropertyExt.prototype.publishedProperties = function (includePrivate, expandProxies) {
         var retVal = [];
         for (var key in this) {
             if (isMeta(key) && (includePrivate || !isPrivate(this, key))) {
-                retVal.push(this[key]);
+                var meta = this[key];
+                if (expandProxies && meta.type) {
+                    var item = this;
+                    while (meta.type === "proxy") {
+                        item = item[meta.proxy];
+                        meta = item.publishedProperty(meta.method);
+                    }
+                    if (meta.id !== this[key].id) {
+                        meta = JSON.parse(JSON.stringify(meta));  //  Clone meta so we can safely replace the id.
```

Comment:
I think this will cause problems with "function" and potentially "array"/"object" publish parameters.